sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFIOWithCassandraTest.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.beam.sdk.io.hadoop.inputformat;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.cassandraunit.utils.EmbeddedCassandraServerHelper;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.runners.MethodSorters;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.mapping.annotations.Column;
+import com.datastax.driver.mapping.annotations.Table;
+
+/**
+ *
+ * Tests to validate HadoopInputFormatIO for embedded Cassandra instance.
+ *
+ */
+@RunWith(JUnit4.class)
+@FixMethodOrder(MethodSorters.JVM)
+public class HIFIOWithCassandraTest implements Serializable {
+	private static final long serialVersionUID = 1L;
+	private static final String CASSANDRA_KEYSPACE = "hif_keyspace";
+	private static final String CASSANDRA_HOST = "127.0.0.1";
+	private static final String CASSANDRA_TABLE = "person";
+	private static transient Cluster cluster;
+	private static transient Session session;
+	@Rule
+	public final transient TestPipeline p = TestPipeline.create();
+
+	@BeforeClass
+	public static void startEmbeddedCassandra() throws Exception {
+		EmbeddedCassandraServerHelper.startEmbeddedCassandra("/cassandra.yaml",
+				"target/cassandra", Long.MAX_VALUE);
```

Comment:
Thanks for pointing this put Stephen. The default time out in EmbeddedCassandraServerHelper.java is 10 sec. If we use this, the test is getting timed out. But 2min timeout is working. Hence, we have set the timeout to 2min as per your suggestion.